src/textual/widget.py

```diff
@@ -1760,6 +1765,9 @@ def _scroll_to(
                 self.scroll_target_y = self.scroll_y = y
                 scrolled_y = scroll_y != self.scroll_y
 
+            if on_complete is not None:
```

Comment:
I think you will need `self.call_after_refresh(on_complete)` here which will allow coroutines to be used, and will ensure the update is called after the screen has updated.

Reply:
fixed! Thanks for going through it.